packages/@o3r/extractors/src/core/comparator/metadata-comparison.helper.ts

```diff
@@ -0,0 +1,135 @@
+
+      if (!migrationMetadataValue) {
+        errors.push(new Error(`Property ${comparator.getName(lastValue)} has been modified but is not documented in the migration document`));
+        continue;
+      }
+
+      if (migrationMetadataValue.after) {
+        const isNewValueInNewMetadata = newMetadataArray.some((newValue) => comparator.isMigrationDataMatch(newValue, migrationMetadataValue.after));
+        if (!isNewValueInNewMetadata) {
+          errors.push(new Error(`Property ${comparator.getName(lastValue)} has been modified but the new property is not present in the new metadata`));
+          continue;
+        }
+      }
+    }
+  }
+  return errors;
+}
+
+/**
+ * Gets the package manager to use to retrieve the previous package from npm.
+ * If the project uses npm or yarn it will be npm.
+ * If the project uses yarn 2+ it will be yarn.
+ * This is especially important because npm and yarn 1 use the authentication from the .npmrc while yarn 2+ uses the .yarnrc.
+ * @param options Option to determine the final package manager
+ */
+function getPackageManagerForRegistry(options?: PackageManagerOptions): SupportedPackageManagers | undefined {
+  const packageManagerInfo = getPackageManagerInfo(options);
+  if (!packageManagerInfo.version) {
+    return undefined;
+  }
+  return packageManagerInfo.name === 'yarn' && !packageManagerInfo.version.match(/^1\./) ? 'yarn' : 'npm';
+}
+
+/**
+ * Checks a type of metadata against a previous version of these metadata extracted from a npm package.
+ * Will return errors if some changes are breaking and they are not allowed, of if the changes are not documented in the file
+ * provided in options.
+ * @param options Options for the buidler
+ * @param context Builder context (from another builder)
+ * @param comparator Comparator implementation, depends on the type of metadata to check
+ */
+export async function checkMetadataBuilder<T>(options: MigrationMetadataCheckBuilderOptions, context: BuilderContext, comparator: MetadataComparator<T>): Promise<BuilderOutput> {
+  context.reportRunning();
+  const angularJsonPath = join(context.workspaceRoot, 'angular.json');
+  const angularJson = existsSync(angularJsonPath) ? JSON.parse(readFileSync(angularJsonPath, { encoding: 'utf8' }).toString()) as WorkspaceSchema : undefined;
+  if (!angularJson) {
+    context.logger.warn(`angular.json file cannot be found by @o3r/core:${context.builder.builderName} builder.
+Detection of package manager runner will fallback on the one used to execute the actual command.`);
+  }
+
+  const packageManager = getPackageManagerForRegistry({
+    workspaceConfig: angularJson,
+    enforcedNpmManager: options.packageManager
+  });
+
+  if (!packageManager) {
+    return {
+      success: false,
+      error: `The package manager to use could not be determined by @o3r/core:${context.builder.builderName}. Try to override it using the packageManager option.`
```

Comment:
is the builder always in @o3r/core ?

Reply:
`context.builder.builderName` does not contain the package name
but `context.builder.name` does, that's what we use in @o3r/telemetry